### PR TITLE
Update Forter JS implementation

### PR DESCRIPTION
**Pull request by ismaeldcom**
Fixes 515-gh-Automattic/woocommerce-payments-server

#### Changes proposed in this Pull Request
- Update Forter JS snippet to the latest version (v13).
- Use the new Forter TTI, that dispatch an event with the token instead of relay on cookies. Add a new rest controller with an endpoint to forward the token to the Fraud Service.
- Update Forter Service to also accept the token as parameter, keeping cookies solution as fallback.
- Disable Forter JS snippet load when the token has been sent.

#### Testing instructions
This depends on the server branch `add/forter-integration`. Be sure to configure the server as mentioned in 1054-gh-Automattic/woocommerce-payments-server. And clear the cache to get Forter included in account `fraud_services`.
- You should see the network request  `.../fraud/forter_token` with the token as payload, on any WC Pay page.
- Check that Forter Service `send_forter_token` is called with both cookie and JS event.
- JS snippet should not load if you set the option `wcpay_forter_token_sent` to your stripe account id.

-------------------

- [x] Added changelog entry (or does not apply)
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

### CODE REVIEW

File: includes/admin/class-wc-rest-payments-fraud-controller.php [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+			]
+		);
+	}
+
+	/**
+	 * Forward token to Fraud Service.
+	 *
+	 * @param WP_REST_Request $request Full data about the request.
+	 */
+	public function send_foter_token( $request ) {
```
Comment (oaratovskyi) [inline includes/admin/class-wc-rest-payments-fraud-controller.php:46]:
```suggestion
	public function send_forter_token( $request ) {
```

File: includes/admin/class-wc-rest-payments-fraud-controller.php [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+	/**
+	 * Configure REST API routes.
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/forter_token',
+			[
+				'methods'             => WP_REST_Server::CREATABLE,
+				'callback'            => [ $this, 'send_foter_token' ],
```
[the remark this answers is not in the document]
Comment (ismaeldcom) [reply includes/admin/class-wc-rest-payments-fraud-controller.php]: 🤦 Thanks @oaratovskyi! Fixed all of them in 6367c56

File: includes/admin/class-wc-rest-payments-fraud-controller.php [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/forter_token',
+			[
+				'methods'             => WP_REST_Server::CREATABLE,
+				'callback'            => [ $this, 'send_forter_token' ],
+				'permission_callback' => [ $this, 'check_permission' ],
+				'args'                => [
+					'token' => [
```
Comment (oaratovskyi) [inline includes/admin/class-wc-rest-payments-fraud-controller.php:33]: I wonder if the token has some minimal string length or so, we might add the arg check here to be a bit more sure about the value we are passing to the fraud service

Comment (ismaeldcom) [reply includes/admin/class-wc-rest-payments-fraud-controller.php:33]: That's a good point. The only thing they mention in API docs is the max length of 128 characters. Other than that, we don't know the allowed characters, or min length. Can you think of any additional check?

**Comment by oaratovskyi** [comment]
@ismaeldcom I tested by following instructions, everything works as it should!
The only thing I noticed is that the client sends two requests at a time
image

File: includes/admin/class-wc-rest-payments-fraud-controller.php (hunk 3 above)
Comment (oaratovskyi) [reply includes/admin/class-wc-rest-payments-fraud-controller.php:33]: Thank you for taking a second look at the docs!
I think in that case it's enough to check that the length is max 128 characters 👍

**Review by oaratovskyi — APPROVED** [review]
LGTM from my perspective! 
Tested it locally, checked that it does not send requests after the option is set. 
The behavior with two requests at a time is not reproduced anymore 🤞

**Pull request merged**